mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -463,10 +462,158 @@ void vector::MultiDimReductionOp::build(OpBuilder &builder,
   build(builder, result, kind, source, acc, reductionDims);
 }
 
+/// Computes the result of reducing a constant vector where the accumulator
+/// value, `acc`, is also constant. `times` is the number of times the operation
+/// is applied.
+template <typename T>
+static OpFoldResult computeConstantReduction(T src, T acc, int64_t times,
```

Comment:
I don't think this needs to be a templates, we should be able to overload this function for all the supported types.